app/controllers/static_controller.rb

```
@@ -27,9 +27,40 @@ class StaticController < ApplicationController
   }
   CUSTOM_PAGES = {} # Add via `#add_topic_static_page` in plugin API
 
+  def validate_redirect_param
+    redirect_path = params[:redirect]
+    if redirect_path.present?
+      raise Discourse::InvalidParameters.new(:redirect) unless redirect_path.is_a?(String)
```

@Grubba27 pointed out that this line might not be necessary. 

I left it in when refactoring the validation logic out of the `enter` method below, but presumably without this line if `params[:redirect]` were a non-string object we'd just eat an error from URI(redirect_path) and return "/".

---

What happens to the user when the `redirect` query parameter is not a string? Do they see a useful error message? Do they need to? Would it be better to "swallow" the error and "ignore" any non-valid `redirect` query parameter? 🤔

---

My reasoning was `Do they need to?` We expect only our client(frontend) to use this feature, right?

---

With the `raise`, the user sees a "You supplied invalid parameters to the request: redirect" err message, only if they somehow get a non-string object as the redirect queryParam. I tested by `/login?redirect[0]=test`.

My gut says this isn't really all that informative and just returning "/" would be better, especially if we don't give an error when the param is a malformed string. 

Maybe logging a warning would be helpful for troubleshooting though, in this case _and_ the case a few lines down where we eat the error trying to convert the string to a URI?

---

I got rid of the Discourse::InvalidParameters error and simply return "/" if the parameter is anything other than a valid string now.